### PR TITLE
OATML Diabetic Retinopathy Benchmark

### PR DESCRIPTION
**Pull request by nband**
Add the RETINA Benchmark introduced in our whitepaper [here](https://openreview.net/forum?id=jyd4Lyjr2iB).

### CODE REVIEW

File: baselines/diabetic_retinopathy_detection/deterministic_tune_on_retention_auc.py [hunk 1]
```diff
@@ -114,6 +115,17 @@ def main(argv):
   logging.info('Saving checkpoints at %s', FLAGS.output_dir)
   tf.random.set_seed(FLAGS.seed)
 
+  # Log Run Hypers
```
Comment (nband) [inline baselines/diabetic_retinopathy_detection/deterministic_tune_on_retention_auc.py:118]: Helpful to assure no xmanager issues with argument setting

File: baselines/diabetic_retinopathy_detection/deterministic_tune_on_retention_auc.py [hunk 2]
```diff
@@ -39,7 +40,7 @@
     'avoid overwriting.')
 flags.DEFINE_string('data_dir', None, 'Path to training and testing data.')
 flags.DEFINE_bool('use_validation', True, 'Whether to use a validation split.')
-flags.DEFINE_bool('use_test', True, 'Whether to use a test split.')
+flags.DEFINE_bool('use_test', False, 'Whether to use a test split.')
```
Comment (nband) [inline baselines/diabetic_retinopathy_detection/deterministic_tune_on_retention_auc.py:43]: Disable test split by default

File: baselines/diabetic_retinopathy_detection/deterministic_tune_on_retention_auc.py [hunk 3]
```diff
@@ -290,11 +303,11 @@ def step_fn(inputs):
     logging.info(message)
 
     # Run evaluation on all evaluation datasets, and compute metrics
-    total_results = utils.evaluate_model_and_compute_metrics(
+    per_pred_results, total_results = utils.evaluate_model_and_compute_metrics(
```
Comment (nband) [inline baselines/diabetic_retinopathy_detection/deterministic_tune_on_retention_auc.py:306]: We cache per-prediction results

File: baselines/diabetic_retinopathy_detection/dropout_tune_on_retention_auc.py [hunk 4]
```diff
@@ -48,6 +49,7 @@
    "'moderate': classify {0, 1} vs {2, 3, 4}, i.e., moderate DR or worse?"))
 flags.DEFINE_bool(
   'load_from_checkpoint', False, "Attempt to load from checkpoint")
+flags.DEFINE_bool('cache_eval_datasets', False, 'Caches eval datasets.')
```
Comment (nband) [inline baselines/diabetic_retinopathy_detection/dropout_tune_on_retention_auc.py:52]: Add this flag, but set to False by default

File: baselines/diabetic_retinopathy_detection/dropout_tune_on_retention_auc.py [hunk 5]
```diff
@@ -145,7 +158,9 @@ def main(argv):
   test_splits = [split for split in available_splits if 'test' in split]
   eval_splits = [split for split in available_splits
                  if 'validation' in split or 'test' in split]
-  eval_datasets = {split: datasets[split] for split in eval_splits}
+
+  # Iterate eval datasets
+  eval_datasets = {split: iter(datasets[split]) for split in eval_splits}
```
Comment (nband) [inline baselines/diabetic_retinopathy_detection/dropout_tune_on_retention_auc.py:163]: We will be repeating eval datasets, so we can just run iter() once here, and not later

File: baselines/diabetic_retinopathy_detection/dropout_tune_on_retention_auc.py [hunk 6]
```diff
@@ -298,11 +313,12 @@ def step_fn(inputs):
     logging.info(message)
 
     # Run evaluation on all evaluation datasets, and compute metrics
-    total_results = utils.evaluate_model_and_compute_metrics(
+    per_pred_results, total_results = utils.evaluate_model_and_compute_metrics(
       strategy, eval_datasets, steps, metrics, eval_estimator,
       uncertainty_estimator_fn, per_core_batch_size, available_splits,
-      estimator_args=estimator_args, is_deterministic=False,
-      num_bins=FLAGS.num_bins, use_tpu=use_tpu)
+      estimator_args=estimator_args, call_dataset_iter=False,
```
Comment (nband) [inline baselines/diabetic_retinopathy_detection/dropout_tune_on_retention_auc.py:319]: Note that we set call_dataset_iter=False here, to reflect that the datasets are already iters

**Review by qixuanf — COMMENTED** [review]
@googlebot I consent.

**Review by qixuanf — APPROVED** [review]
@googlebot I consent.

**Review by qixuanf — COMMENTED** [review]
@googlebot I consent.

**Review by znado — APPROVED** [review]
Approving for now so we can review internally (please do not merge this yet!)

**Pull request closed without merging**